passes/S038/testdata/src/a/main.go

```diff
@@ -0,0 +1,90 @@
+package a
+
+import "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+
+func f() {
+	// Comment ignored
+
+	//lintignore:S038
+	_ = &schema.Resource{
+		Read: func(*schema.ResourceData, interface{}) error { return nil },
+		Schema: map[string]*schema.Schema{
+			"x": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"foo": {
+							AtLeastOneOf: []string{"x.0.bar"},
+						},
+						"bar": {
+							AtLeastOneOf: []string{"x.0.foo"},
+						},
+					},
+				},
+			},
+		},
+	}
+	// Failing
+
+	_ = &schema.Resource{
+		Read: func(*schema.ResourceData, interface{}) error { return nil },
+		Schema: map[string]*schema.Schema{
+			"x": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"foo": {
+							AtLeastOneOf: []string{"x.0.bar"}, // want `S038: invalid AtLeastOneOf attribute reference semantics: "x.0" configuration block attribute references are only valid for TypeList and MaxItems: 1 attributes`
+						},
+						"bar": {
+							AtLeastOneOf: []string{"x.0.foo"}, // want `S038: invalid AtLeastOneOf attribute reference semantics: "x.0" configuration block attribute references are only valid for TypeList and MaxItems: 1 attributes`
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_ = &schema.Resource{
+		Read: func(*schema.ResourceData, interface{}) error { return nil },
+		Schema: map[string]*schema.Schema{
+			"x": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"foo": {
+							AtLeastOneOf: []string{"x.1.bar"}, // want `S038: invalid AtLeastOneOf attribute reference semantics: "x.1" configuration block attribute references must be separated by .0`
+						},
+						"bar": {
+							AtLeastOneOf: []string{"x.1.foo"}, // want `S038: invalid AtLeastOneOf attribute reference semantics: "x.1" configuration block attribute references must be separated by .0`
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Passing
```

Comment:
Let's ensure the following additional passing cases are tested:

- Root attribute referencing existing root attribute (e.g. https://github.com/terraform-providers/terraform-provider-aws/blob/2cb4e3f7f017feb338e0ba2ffdf409d6b0c91793/aws/resource_aws_cognito_user_pool.go#L90)
- Root attribute referencing existing configuration block attribute with existing nested attribute (e.g. https://github.com/terraform-providers/terraform-provider-aws/blob/2cb4e3f7f017feb338e0ba2ffdf409d6b0c91793/aws/resource_aws_cognito_user_pool.go#L176)
- Configuration block nested attribute referencing existing root attribute (e.g. https://github.com/terraform-providers/terraform-provider-aws/blob/2cb4e3f7f017feb338e0ba2ffdf409d6b0c91793/aws/resource_aws_cognito_user_pool.go#L517)
- Configuration block nested attribute referencing existing configuration block attribute with existing nested attribute (e.g. https://github.com/terraform-providers/terraform-provider-aws/blob/2cb4e3f7f017feb338e0ba2ffdf409d6b0c91793/aws/resource_aws_cognito_user_pool.go#L307)

Reply:
Done